pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java

```diff
@@ -122,6 +122,13 @@ private CompletableFuture<Boolean> checkDlqAlreadyExists(String topic) {
                     || actEx instanceof PulsarClientException.TopicDoesNotExistException
                     || actEx instanceof PulsarAdminException.NotFoundException) {
                 existsFuture.complete(false);
+            } else if (actEx instanceof PulsarClientException.NotSupportedException) {
+                existsFuture.completeExceptionally(new PulsarClientException.NotSupportedException("There is a bug that"
+                    + " the Retry/DLQ consumer will still trigger a Retry/DLQ topic with the old rule"
+                    + " ({namespace}/{subscription}-RETRY/DLQ), but the rule was changed to"
+                    + " {namespace}/{topic}-{subscription}-RETRY/DLQ after 2.8.0. Please upgrade the brokers' version"
+                    + " to >=3.0.6 or >=3.3.1; another solution is use HTTP protocol service URL instead of Binary"
+                    + " protocol service URL when building Pulsar Client"));
```

Comment:
Instead of vaguely referring to "a bug", would it be possible to add the URL of the issue report?
It feels a bit personal when the log message contains "please upgrade the broker's version to ...". 
Since the fix might be included later to other versions, it might be better to be brief about the resolution. "The resolution is to upgrade to a broker version with fix (PR link) included. A workaround is to switch to use http based service url for the Pulsar client." etc.

Reply:
Changed the implementation, please review gain